build/_build/Build.GitHub.cs

```diff
@@ -377,90 +380,179 @@ partial class Build
-                                    $"Are you sure you've merged the version bump PR?");
-            }
+            var resourceDownloadUrl = artifact.Resource.DownloadUrl;
+            Console.WriteLine("::set-output name=artifacts_link::" + resourceDownloadUrl);
+            Console.WriteLine("::set-output name=artifacts_path::" + OutputDirectory / artifact.Name);
+        });
 
-            var completedBuilds = builds
-                                 .Where(x => x.Status == BuildStatus.Completed)
-                                 .ToList();
-            if (!completedBuilds.Any())
-            {
-                throw new Exception($"Error: no completed builds for {branch} were found. " +
-                                    $"Please wait for completion before running this workflow.");
-            }
+    Target CompareCodeCoverageReports => _ => _
+         .Unlisted()
+         .DependsOn(CreateRequiredDirectories)
+         .Requires(() => AzureDevopsToken)
+         .Requires(() => GitHubToken)
+         .Executes(async () =>
+          {
+              var newReportdir = OutputDirectory / "CodeCoverage" / "New";
+              var oldReportdir = OutputDirectory / "CodeCoverage" / "Old";
+
+              FileSystemTasks.EnsureCleanDirectory(newReportdir);
+              FileSystemTasks.EnsureCleanDirectory(oldReportdir);
+
+              // Connect to Azure DevOps Services
+              var connection = new VssConnection(
+                  new Uri(AzureDevopsOrganisation),
+                  new VssBasicCredential(string.Empty, AzureDevopsToken));
+
+              // Get a GitHttpClient to talk to the Git endpoints
+              using var buildHttpClient = connection.GetClient<BuildHttpClient>();
+
+              var prNumber = int.Parse(Environment.GetEnvironmentVariable("PR_NUMBER"));
+              var branch = $"refs/pull/{prNumber}/merge";
+              var fixedPrefix = "Code Coverage Report_";
+
+              var (newBuild, newArtifact) = await DownloadAzureArtifact(buildHttpClient, branch, build => $"{fixedPrefix}{build.Id}", newReportdir, buildReason: null, completedBuildsOnly: false);
+              var (oldBuild, oldArtifact) = await DownloadAzureArtifact(buildHttpClient, "refs/heads/master", build => $"{fixedPrefix}{build.Id}", oldReportdir, buildReason: null);
+
+              var oldBuildId = oldArtifact.Name.Substring(fixedPrefix.Length);
+              var newBuildId = newArtifact.Name.Substring(fixedPrefix.Length);
+
+              var oldReportPath = oldReportdir / oldArtifact.Name / $"summary{oldBuildId}" / "Cobertura.xml";
+              var newReportPath = newReportdir / newArtifact.Name / $"summary{newBuildId}" / "Cobertura.xml";
+
+              var downloadOldLink = oldArtifact.Resource.DownloadUrl;
+              var downloadNewLink = newArtifact.Resource.DownloadUrl;
+
+              // var oldReportPath = RootDirectory / "Cobertura.old.xml";
```

Comment:
Debug comments?

Reply:
Ah, yes and no - I was using this to compare two _local_ coverage files. That might be useful enough to separate out into a dedicated target 🤔

Reply:
Fixed in https://github.com/DataDog/dd-trace-dotnet/pull/1704/commits/8a7db0e5b58679872b803b47c25434462ba93783